prefect_shell/commands.py

```diff
@@ -60,12 +60,17 @@ def example_shell_run_command_flow():
     current_env = os.environ.copy()
     current_env.update(env or {})
 
-    with tempfile.NamedTemporaryFile(prefix="prefect-") as tmp:
+    if shell.lower() == "powershell":
+        suffix = ".ps1"
+    else:
+        suffix = ""
+
+    with tempfile.NamedTemporaryFile(prefix="prefect-", suffix=suffix, delete=False) as tmp:
```

Comment:
```suggestion
    with tempfile.NamedTemporaryFile(
        prefix="prefect-", suffix=suffix, delete=False
    ) as tmp:
```

Reply:
Or since `delete = False`, maybe this can be
`tmp = tempfile.NamedTemporaryFile(prefix="prefect-", suffix=suffix, delete=False)`
to call out that this context manager does not delete after the fact.